rest/diagnostic_doc_api.go

```diff
@@ -39,3 +54,53 @@ func (h *handler) handleGetDocChannels() error {
 	h.writeJSON(resp)
 	return nil
 }
+
+// HTTP handler for running a document through the sync function and returning the results
+func (h *handler) handleSyncFnDryRun() error {
+	docid := h.getQuery("docid")
+	revID := h.getQuery("rev")
```

Comment:
Requiring callers to provide a revID for a dry run seems like unnecessary complexity.  The main purpose of exposing revID in the rest of the REST API generally is as a mechanism for optimistic concurrency, which we don't care about for this endpoint.  

I understand that we inject revID into doc and oldDoc when running the sync function, so it may make sense to populate those for proper sync function evaluation.  Can we populate revID values ourselves?  (i.e. treat as an insert if the document doesn't exist, create as a child if it does).  This will simplify future enhancements to work with VV-based revIDs as well.

Reply:
Fixed